### PR TITLE
feat(lanelet2_map_preprocessor): port lanelet2_map_preprocessor from ROS1 to ROS2

### PR DESCRIPTION
**Pull request by mitsudome-r**
## Related Issue(required)

part of 
https://github.com/autowarefoundation/autoware.universe/issues/457
## Description(required)

This ports lanelet2_map_preprocessor package into ROS2. 

## Review Procedure(required)
Run the following launch files and make sure that it works:
* `ros2 launch lanelet2_map_preprocessor transform_maps.launch.xml llt_map_path:=<input_lanelet2_map> pcd_map_path:=<input pointcloud_map> llt_output_path:=<output_lanelet2_map> pcd_output_path:=<output_pointcloud_map> x:=<translation value for x>`
* ros2 launch lanelet2_map_preprocessor fix_z_value_by_pcd.launch.xml  llt_map_path:=<input_lanelet2_map> pcd_map_path:=<input pointcloud_map> llt_output_path:=<output_lanelet2_map>`

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [ ] Read [commit-guidelines][commit-guidelines]
- [ ] Assign PR to reviewer

If you are adding new package following items are required:

- [ ] Documentation with description of the package is available
- [ ] A sample launch file and parameter file are available if the package contains executable nodes

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[commit-guidelines]: https://www.conventionalcommits.org/en/v1.0.0/
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

File: map/util/lanelet2_map_preprocessor/CMakeLists.txt [hunk 1]
```diff
@@ -1,56 +1,43 @@
...
+# add_dependencies(merge_close_points ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})
+# target_link_libraries(merge_close_points ${catkin_LIBRARIES})
+
+# add_executable(remove_unreferenced_geometry src/remove_unreferenced_geometry.cpp)
+# add_dependencies(remove_unreferenced_geometry ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})
+# target_link_libraries(remove_unreferenced_geometry ${catkin_LIBRARIES})
+
+# add_executable(fix_lane_change_tags src/fix_lane_change_tags.cpp)
+# add_dependencies(fix_lane_change_tags ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})
+# target_link_libraries(fix_lane_change_tags ${catkin_LIBRARIES})
```
Comment (yukkysaito) [inline map/util/lanelet2_map_preprocessor/CMakeLists.txt:34]: Does it mean that some nodes have been ported and some still need to be done?

**Comment by wep21** [comment]
@kenji-miyake closed by #847?

**Comment by kenji-miyake** [comment]
@wep21 Oh, I didn't notice this. I'll rebase it and check the diff.

File: map/util/lanelet2_map_preprocessor/CMakeLists.txt (hunk 1 above)
Comment (kenji-miyake) [reply map/util/lanelet2_map_preprocessor/CMakeLists.txt]: Resolved by #847.

**Review by kenji-miyake — APPROVED** [review]

**Comment by kenji-miyake** [comment]
I'll merge this for Rolling/Humble cleanup.
If there are some problems left, please open a new pull request. :pray:

**Review by kenji-miyake — APPROVED** [review]

**Pull request merged**